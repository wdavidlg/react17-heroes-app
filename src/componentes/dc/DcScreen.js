import React from 'react'
import HeroesList from '../heroes/HeroesList'

const DcScreen = () => {
    return (
        <div>
            <h1>DC Comics</h1>
            <hr/>
            <HeroesList publisher='DC Comics'/>
        </div>
    )
}

export default DcScreen
