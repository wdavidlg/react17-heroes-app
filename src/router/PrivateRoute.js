import React from 'react'
import PropTypes from 'prop-types';
import { Redirect, Route } from 'react-router'

const PrivateRoute = ({
    isAuthenticate,
    component: Component,
    ...rest
}) => {
    localStorage.setItem('lastPath', rest.location.pathname);
    
    return (
        <Route {...rest}
            component={(props) => (
                (isAuthenticate)
                    ?<Component {...props}/>
                    :<Redirect to='/login' />
            )}
        />
    )
}
PrivateRoute.propTypes = {
    isAuthenticate: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}

export default PrivateRoute
