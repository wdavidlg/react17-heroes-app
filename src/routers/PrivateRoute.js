import React, { useContext } from 'react'
import { Navigate, useLocation } from 'react-router-dom'
import { AuthContext } from '../auth/AuthContext';

const PrivateRoute = ({children}) => {
    const {user} = useContext(AuthContext);
    const location = useLocation();
    localStorage.setItem('lastPath', location.pathname);
    
    return user.logged ? children : <Navigate to='/login' />
}


export default PrivateRoute
