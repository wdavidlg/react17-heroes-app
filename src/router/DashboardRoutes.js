import React from 'react'
import { Redirect, Route, Switch } from 'react-router'
import DcScreen from '../componentes/dc/DcScreen'
import HeroeScreen from '../componentes/heroes/HeroeScreen'
import MarvelScreen from '../componentes/marvel/MarvelScreen'
import SearchScreen from '../componentes/search/SearchScreen'
import { NavBar } from '../componentes/ui/NavBar'

const DashboardRoutes = () => {


    return (
        <> 
            <NavBar />
            <div className='container mt-5'>
                <Switch>
                    <Route exact path='/marvel' component={MarvelScreen} />
                    <Route exact path='/heroe/:heroeId' component={HeroeScreen} />
                    <Route exact path='/dc' component={DcScreen} />
                    <Route exact path='/search' component={SearchScreen} />
                    <Redirect to='/marvel' />
                </Switch>
            </div>
        </>
    )
}

export default DashboardRoutes
