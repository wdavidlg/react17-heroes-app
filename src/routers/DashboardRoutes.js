import React from 'react'
import { Navigate, Route, Routes} from 'react-router-dom'
import DcScreen from '../componentes/dc/DcScreen'
import HeroeScreen from '../componentes/heroes/HeroeScreen'
import MarvelScreen from '../componentes/marvel/MarvelScreen'
import SearchScreen from '../componentes/search/SearchScreen'
import { NavBar } from '../componentes/ui/NavBar'

const DashboardRoutes = () => {


    return (
        <> 
            <NavBar />
            <div className='container mt-5'>
                <Routes>
                    <Route path='/marvel' element={<MarvelScreen />} />
                    <Route path='/heroe/:heroeId' element={<HeroeScreen />} />
                    <Route path='/dc' element={<DcScreen />} />
                    <Route path='/search' element={<SearchScreen />} />
                    <Route path='/*' element={<Navigate to='/marvel' />} />
                </Routes>
            </div>
        </>
    )
}

export default DashboardRoutes
