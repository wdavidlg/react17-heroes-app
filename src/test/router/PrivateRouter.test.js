import {mount, shallow} from 'enzyme'
import { MemoryRouter } from 'react-router'
import DcScreen from '../../componentes/dc/DcScreen'
import DashboardRoutes from '../../router/DashboardRoutes'
import PrivateRoute from '../../router/PrivateRoute'

describe('Pruebas en <PrivateRoute />', () => {
    
    test('debe de mostrar el componente si esta autenticado y guardar localStorage', () => {

        const wrapper = mount(
            <MemoryRouter>
                <PrivateRoute 
                isAuthenticate={true}
                component={()=><span>Listo</span>}
                location = {{pathname:'/dc'}}
                />
            </MemoryRouter>
        )
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('span').exists()).toBe(true);
       // expect(wrapper).toMatchSnapshot();
    })

    test('debe de llamar a localStorage', () => {

        Storage.prototype.setItem = jest.fn();

        const wrapper = mount(
            <MemoryRouter>
                <PrivateRoute 
                isAuthenticate={true}
                component={()=><span>Listo</span>}
                location = {{pathname:'/dc'}}
                />
            </MemoryRouter>
        )
        
        expect(localStorage.setItem).toHaveBeenCalledWith("lastPath", "/dc");
       
    })

    test('debe de bloquear el componente si no esta autenticado', () => {
        const wrapper = mount(
            <MemoryRouter>
                <PrivateRoute 
                isAuthenticate={false}
                component={()=><span>Listo</span>}
                location = {{pathname:'/dc'}}
                />
            </MemoryRouter>
        )
        
        console.log(wrapper.html())
        expect(wrapper.html()).toBe("")
        
    })
    
    
    
    
})
