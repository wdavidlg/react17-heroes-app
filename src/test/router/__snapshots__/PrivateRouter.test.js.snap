// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrivateRoute /> debe de mostrar el componente si esta autenticado y guardar localStorage 1`] = `
<span>
  Listo
</span>
`;
